### PR TITLE
mariadb: 10.4.13 -> 10.4.14

### PR DESCRIPTION
**Pull request by Izorkin**
###### Motivation for this change
Update to version 10.4.14

cc @thoughtpolice @flokli @aanderse @dasJ

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mweinelt** [comment]
@GrahamcOfBorg test mariadb-galera-mariabackup mariadb-galera-rsync mysql mysql-autobackup mysql-backup mysql-replication

**Comment by Izorkin** [comment]
All tests fail for an unknown reason:
```
patching /nix/store/1jrw092qwwgw5wkf4sqf3qmn9f2y7jbl-extra-utils/bin/scsi_id...
patching /nix/store/1jrw092qwwgw5wkf4sqf3qmn9f2y7jbl-extra-utils/bin/blkid...
testing patched programs...
hello world
245
mdadm - v4.1 - 2018-10-01
-- Looking for dlopen in dl - found
-- Performing Test have_C__ggdb3
output '/nix/store/1jrw092qwwgw5wkf4sqf3qmn9f2y7jbl-extra-utils' is not allowed to refer to the following paths:
  /nix/store/v589pqjhvxrj73g3r0xb41yr84z5pwb7-gcc-9.3.0-lib
cannot build derivation '/nix/store/9paiybhyc41vgwga8f2jzfwchkqifcj8-stage-1-init.sh.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/6x74q96jpmanw0d2m0yq3yzzkwk40av4-initrd-linux-5.4.57.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/jc4ypavw8hnz5jpln10wg1cvwkdc7bgm-nixos-system-mariadb-20.09.git.abd12784443.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/wxmyjhi9xyhbaw68c4wbamgkpzaqm96g-nixos-system-mysql57-20.09.git.abd12784443.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/w56jq1pqb7fdp2y0vfbl4z7dzxaqjai4-nixos-system-mysql80-20.09.git.abd12784443.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/hmvidl3z8g3nbqc1fibdq5kcmsr2xjgz-nixos-vm.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/hqdicp5ma0pd1nbbrwamzba32mwr7gjz-nixos-vm.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/rm75cgvrrsb8q8wfc5v70jrnak7fba28-nixos-vm.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/ijbhhjcxs2i3yksdns3m4s8n1cx8jfw8-nixos-test-driver-mysql.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/g6nw5ahsa4lbh1jlksy9l731swkw2rb2-vm-test-run-mysql.drv': 1 dependencies couldn't be built
error: build of '/nix/store/g6nw5ahsa4lbh1jlksy9l731swkw2rb2-vm-test-run-mysql.drv' failed
```

**Comment by flokli** [comment]
This is fixed in master, you need to rebase.

**Comment by Izorkin** [comment]
@GrahamcOfBorg test mariadb-galera-mariabackup mariadb-galera-rsync mysql mysql-autobackup mysql-backup mysql-replication

**Comment by Izorkin** [comment]
@GrahamcOfBorg test mysql

**Comment by Izorkin** [comment]
@GrahamcOfBorg test mysql

**Comment by Izorkin** [comment]
@GrahamcOfBorg test mariadb-galera-mariabackup
@GrahamcOfBorg test mariadb-galera-rsync
@GrahamcOfBorg test mysql
@GrahamcOfBorg test mysql-autobackup
@GrahamcOfBorg test mysql-backup
@GrahamcOfBorg test mysql-replication

**Comment by Izorkin** [comment]
@GrahamcOfBorg eval

File: nixos/tests/mysql/mysql.nix [hunk 1]
```diff
@@ -172,32 +172,32 @@ import ./../make-test-python.nix ({ pkgs, ...} : {
...
-        "echo 'use testdb; select test_id from tokudb;' | sudo -u testuser mysql -u testuser -N | grep 25"
+        "echo 'use testdb; select test_id from rocksdb;' | sudo -u testuser mysql -u testuser -N | grep 28"
     )
     mariadb.succeed(
-        "echo 'use testdb; drop table tokudb;' | sudo -u testuser mysql -u testuser"
+        "echo 'use testdb; drop table rocksdb;' | sudo -u testuser mysql -u testuser"
     )
-
-    # Check if RocksDB plugin works
+  '' + pkgs.stdenv.lib.optionalString (!pkgs.stdenv.hostPlatform.isAarch32 && !pkgs.stdenv.hostPlatform.isAarch64) ''
```
Comment (flokli) [inline nixos/tests/mysql/mysql.nix:188]: We should better check for architectures supported, not those not supported - TokuDB probably also doesn't run on riscv5 ;)

```suggestion
  '' + pkgs.stdenv.lib.optionalString pkgs.stdenv.isx86_64 ''
```

Also, shouldn't this be pkgs.stdenv.

```suggestion
  '' + pkgs.stdenv.lib.optionalString (!pkgs.stdenv.hostPlatform.isAarch32 && !pkgs.stdenv.hostPlatform.isAarch64) ''
```

Comment (Izorkin) [reply nixos/tests/mysql/mysql.nix]: Fixed.

**Comment by Izorkin** [comment]
@GrahamcOfBorg eval

**Pull request merged**